### ISSUE TITLE
Adding use examples in synthetic functions

### ISSUE THREAD

**Issue by DavidSiretMarques** [pytorch/botorch#2039]
# 🚀 Feature Request: Add use examples in synthetic functions

Example uses for functions in  `botorch.test_functions.synthetic`
## Motivation

**Is your feature request related to a problem? Please describe.**

I'm trying to do some benchmark testing for a project I'm working on, and I decided to understand first how the synthetic function module works and what and how it returns things. But I'm not able to do an Ackley 2-dimensional function plot. I've tried everything I can think about, giving it single values, trying with different sized tensors,... But I don't get how to use this function (or any other function in the module). 

## Pitch

**Describe the solution you'd like**

I think it would be useful for the docs to have a couple of examples for the functions in this module, with how they work and with that it would be easier to understand how to use them.

**Describe alternatives you've considered**

Alternatively, there could be a jupyter notebook where some of these functions are used, to show how they work

**Are you willing to open a pull request?** (See [CONTRIBUTING](../CONTRIBUTING.md))
Sure, If I can understand how the function works...

**Comment by Balandat** [comment]
Hey, sorry you're having trouble with this. 

Not sure where you got stuck, but here is an example for how to use 2d Ackley: 
```
import torch

from botorch.test_functions import Ackley


ack = Ackley(dim=2)

n = 5  # number of random locations to evaluate on

# Ackley is usually evaluated on [-32.768, 32.768]^d so we scale X to that
X = ack.bounds[0] + torch.rand(5, 2) * (ack.bounds[1] - ack.bounds[0])

f = ack(X)

f
tensor([21.0521, 19.9780, 20.2577, 21.0732, 21.8011], dtype=torch.float64)
```

You can also add a `noise_std` if you want the function to produce observations of the form `f(X) + noise_std * eps`, where `eps` is a standard normal RV.

We can add this to the docstring of the test functions module if that would be helpful.

**Comment by DavidSiretMarques** [comment]
I spent a good part of yesterday trying to understand why if I gave Ackley 2 vectors with 5 points each (like `[1,2,3,4,5]`) it only returned 2 numbers, instead of 5.

Just to be clear, I was doing

```python
Ackley([[1,2,3,4,5][1,2,3,4,5]])
```

And this returned only 2 points, instead of the 5 i was expecting... @_@

**Comment by DavidSiretMarques** [comment]
In any case, even if not everything is done, I think you should specify the shape of the tensor to feed the `evaluate_true(X)` method. Saying in the docs that X should have shape dim x n-points would be really helpful to any person trying to use the function (I might have it wrong and the tensor should be the other way around, but I think the point is clear anyway...)
